spark/spark-base/src/main/java/com/thoughtworks/go/spark/Routes.java

```diff
@@ -138,6 +138,11 @@ public static String vsm(String materialFingerprint, String revision) {
         }
     }
 
+    public static class MailHost {
+        public static final String SELF = "/api/admin/mailhost";
+        public static final String DOC = "https://api.go.cd/current/#mailhost";
```

Comment:
Do we want to the endpoint to be `/api/admin/mailhost` or `/api/admin/config/server/mailhost` or `/api/admin/server/mailhost`?

Whatever we choose should be consistent with https://github.com/gocd/gocd/issues/4189# IMO

Reply:
How about `/api/admin/config/{smtp,emailserver,mailserver}`. I don't know why it was called `mailhost`, but my preferene is smtp, followed by emailserver and mailserver.

Reply:
@ketan @akshaydewan -- If we decide to call this API endpoint `smtp` or `emailserver`, does it make sense to change the xml entity from `mailhost` to whatever we decide?

Reply:
Does anyone have opinions on this question?

Reply:
I think `mailserver` makes sense. I wouldn't bother changing it in the XML. It's likely quite an underused property. But, if it makes you happy to keep it consistent, @GaneshSPatil , I can understand. :)

Reply:
@arvindsv -- we are adding an XML migration to move the server attribute into a separate child tag for configurations such as: `Server URLs`, `Pipeline artifacts`, `Job Timeout`, `Command Repository`, etc. Hence thought of renaming this too as part of same migration. :)